### PR TITLE
Add the time when it will expire to debug log

### PR DESCRIPTION
**Pull request by mikutas**
Do we want to know when to expire even if credentials are not expired for now ?

### CODE REVIEW

File: pkg/awsconfig/awsconfig.go [hunk 1]
```diff
@@ -113,13 +113,13 @@ func (p *CredentialsProvider) Load() (*AWSCredentials, error) {
 }
 
 // Expired checks if the current credentials are expired
-func (p *CredentialsProvider) Expired() bool {
+func (p *CredentialsProvider) Expired() (bool, *time.Time) {
```
Comment (sledigabel) [inline pkg/awsconfig/awsconfig.go:116]: The function name doesn't really do what it says now.
Why not create a new method `ExpiresAt` to return the `*time.Time` instead?

File: pkg/awsconfig/awsconfig.go [hunk 2]
```diff
@@ -122,6 +122,15 @@ func (p *CredentialsProvider) Expired() bool {
 	return time.Now().After(creds.Expires)
 }
 
+func (p *CredentialsProvider) ExpiredAt() *time.Time {
```
Comment (sledigabel) [inline pkg/awsconfig/awsconfig.go:125]: I would recommend passing it as value, since a `time.Time` is a measure of Time.

**Review by sledigabel — COMMENTED** [review]
Looks nice 👍  just a small comment on using value semantic

File: pkg/awsconfig/awsconfig.go (hunk 2 above)
Comment (mikutas) [reply pkg/awsconfig/awsconfig.go]: Just noticed :eyes:, can we perhaps use previousCreds.Expires instead of adding ExpiredAt?

Comment (sledigabel) [reply pkg/awsconfig/awsconfig.go]: @mikutas that is indeed a _lot_ simpler :-)

**Comment by sledigabel** [comment]
FWIW, LGTM! 🚀

**Comment by gliptak** [comment]
@mapkon please rebase

**Comment by mapkon** [comment]
> @mapkon please rebase

Do you want me to merge this?

**Comment by gliptak** [comment]
yes please consiider merging. @mikutas found this useful for debugging

**Pull request merged**